kube-core/src/params.rs

```diff
@@ -347,6 +384,7 @@ impl Default for WatchParams {
             label_selector: None,
             field_selector: None,
             timeout: None,
+            send_initial_events: false,
```

Comment:
This is a sensible default for now. I think we should maybe add a new constructor that sets correct flags rather than mess with the builders/serialization. Something like:

```rust
impl WatchParams
    /// Constructor for doing Kubernetes 1.27 Streaming List watches
    ///
    /// Enables [`VersionMatch::NotGreaterThan`] semantics and [`WatchParams::send_initial_events`].
    pub fn streaming_lists() -> Self {
        Self {
            send_initial_events: true,
            version_match: Some(VersionMatch::NotGreaterThan), // required
            bookmarks: true, // required
            ..WatchParams::default()
        }
    }
```